starter-code/database.js

```diff
@@ -24,105 +23,106 @@ class Database {
     }
   }
 
-  close(callback = (error) => {}){
-    if (this.database){
+  close (callback = (error) => {}) {
+    if (this.database) {
       this.database.close(true, callback);
     } else {
       callback();
     }
   }
   // Insert a user
   // user is the object to insert into the collection
-  // callback has two arguments error and result 
-  insertUser(user, callback = (error, result) => {}){
+  // callback has two arguments error and result
+  insertUser (user, callback = (error, result) => {}) {
     this.connect((error, database) => {
-      if (error){
+      if (error) {
         callback(error);
       } else {
         // LAB 1
-        // Implement the query to insert a user
         // user is the document that we want to insert
         // remeber once it's finish to comment callback('Error inserting user');
-        
-        callback('Error inserting user');
+        database.collection('users').insertOne(user, callback);
+
+        // callback('Error inserting user');
       }
     });
   }
 
-  listUsers(callback = (error, users) => {}) {
+  listUsers (callback = (error, users) => {}) {
     this.connect((error, database) => {
-      if (error){
+      if (error) {
         callback(error);
       } else {
         //  LAB 2
         // Implement the query to insert a user
         // remeber once it's finish to comment callback('Error listing users');
-        
-        callback('Error listing users');
+        database.collection('users').find(callback);
+        // callback('Error listing users');
       }
     });
   }
 
-  deleteUser( firstName, callback = (error, result) => {}) {
+  deleteUser (firstName, callback = (error, result) => {}) {
     this.connect((error, database) => {
-      if (error){
+      if (error) {
         callback(error);
       } else {
         //  LAB 3
         // Implement the query to delete a user
         // firstName is the name of user that we want to delete
         // remeber once it's finish to comment callback('Error deleting user');
-        
-        callback('Error deleting user');
+        database.collection('users').remove(firstName, callback);
```

Comment:
This is the same as this:

```
database.collection('users').remove({'firstName' : firstName}, callback);
```

(Maybe you already knew it, but be sure you understand the structure behind this abbreviation)